modules/angular-meteor-meteorCollection.js

```diff
@@ -239,48 +255,25 @@ angularMeteorCollections.factory('$meteorCollection', ['AngularMeteorCollection'
       }
 
       var ngCollection = new AngularMeteorCollection(reactiveFunc(), collection);
-      var realOldItems;
 
       function setAutoBind() {
         if (auto) { // Deep watches the model and performs autobind.
           ngCollection.unregisterAutoBind = $rootScope.$watch(function () {
-            if (ngCollection.UPDATING_FROM_SERVER){
-              realOldItems = _.without(ngCollection, 'UPDATING_FROM_SERVER');
+            if (ngCollection.UPDATING_FROM_SERVER) {
               return 'UPDATING_FROM_SERVER';
             }
             return _.without(ngCollection, 'UPDATING_FROM_SERVER');
           }, function (newItems, oldItems) {
-            if (newItems == 'UPDATING_FROM_SERVER')
+            if (newItems == 'UPDATING_FROM_SERVER' ||
+                oldItems == 'UPDATING_FROM_SERVER')
               return;
 
-            if (oldItems == 'UPDATING_FROM_SERVER')
-              oldItems = realOldItems;
-
-
             if (newItems !== oldItems) {
+              ngCollection.unregisterAutoBind();
+
+              updateCollection(ngCollection, oldItems, diffArray);
```

Comment:
This part seems to have not been changed if you mean ````ngCollection.unregisterAutoBind````, I only wrapped code into updateCollection. I'd fix it in a different PR.